dotCMS/src/main/java/com/dotcms/ema/EMAWebInterceptor.java

```diff
@@ -0,0 +1,173 @@
+        }
+
+        Logger.info(this.getClass(), "GOT AN EMA Call --> " + request.getRequestURI());
+
+        return new Result.Builder().wrap(new MockHttpCaptureResponse(response)).next().build();
+    }
+
+
+    @Override
+    public boolean afterIntercept(final HttpServletRequest request, final HttpServletResponse response) {
+
+        try {
+
+            if (response instanceof MockHttpCaptureResponse) {
+
+                final Host currentHost = WebAPILocator.getHostWebAPI().getCurrentHostNoThrow(request);
+                final Optional<String> proxyUrl            = proxyUrl(currentHost);
+                final MockHttpCaptureResponse mockResponse = (MockHttpCaptureResponse)response;
+                final String postJson                      = new String(mockResponse.getBytes());
+                final JSONObject json                      = new JSONObject(postJson);
+                final Map<String, String> params           = ImmutableMap.of("dotPageData", postJson);
+                
+                Logger.info(this.getClass(), "Proxying Request --> " + proxyUrl.get());
+
+                final StringBuilder responseStringBuilder = new StringBuilder();
+                final ProxyResponse pResponse = proxy.sendPost(proxyUrl.get(), params);
+
+                if (pResponse.getResponseCode() == 200) {
+                    responseStringBuilder.append(pResponse.getResponse());
+                }else {
+                    responseStringBuilder.append("<html><body>")
+                    .append("<h3>Unable to connect with the rendering engine</h3>")
+                    .append("<br><div style='display:inline-block;width:80px'>Trying: </div><b>" + proxyUrl.get()  + "</b>")
+                    .append("<br><div style='display:inline-block;width:80px'>Got:</div><b>" + pResponse.getStatus() + "</b>")
+                    .append("<hr>")
+                    .append("<h4>Headers</h4>")
+                    .append("<table border=1 style='min-width:500px'>");
+
+                    for(final Header header : pResponse.getHeaders()) {
+                        responseStringBuilder.append("<tr><td style='font-weight:bold;padding:5px;'><pre>" + header.getName() + "</pre></td><td><pre>" + header.getValue() + "</td></tr>");
+                    }
+                    responseStringBuilder.append("</table>")
+                    .append("<p>The Json Payload, POSTing as Content-Type:'application/x-www-form-urlencoded' with form param <b>dotPageData</b>, has been printed in the logs.</p>")
+                    .append("</body></html>");
+
+                }
+
+                json.getJSONObject("entity").getJSONObject("page").put("rendered", responseStringBuilder.toString());
+                json.getJSONObject("entity").getJSONObject("page").put("remoteRendered", true);
+                response.setContentType("application/json");
+
+                response.getWriter().write(json.toString());
+            }
+        } catch (Exception e) {
```

Comment:
Codacy found an issue: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5922984)